Framework/CurveFitting/src/Functions/CrystalFieldMultiSpectrum.cpp

```diff
@@ -125,9 +125,6 @@ size_t CrystalFieldMultiSpectrum::getNumberDomains() const {
   if (!m_target) {
     buildTargetFunction();
   }
-  if (!m_target) {
```

Comment:
I think we still need this check and exception for the same reason that @DTasev pointed out for `CrystalFieldFunction.cpp`?